flow/Arena.h

```diff
@@ -101,6 +101,7 @@ class Arena {
 	Arena& operator=(const Arena&);
 	Arena& operator=(Arena&&) noexcept;
 
+	// If additional ArenaBlock is added to p, dependsOn will need to be re-called in order to include the new blocks.
```

Comment:
Instead of saying "re-called", which is probably an anti-pattern. I'd say something like `Adds a reference to memory blocks held by "p" so that they are accessible after "p" is destructed. Note future blocks allocated by "p" are not referenced.`